Deploy more smart pointers in AbortSignal.cpp, BroadcastChannel.cpp, NodeRareData.h, and Range.cpp

rniwa opened the pull request:
#### 97f82b70814b67345cccae647532a24ca210b4d4
<pre>
Deploy more smart pointers in AbortSignal.cpp, BroadcastChannel.cpp, NodeRareData.h, and Range.cpp
<a href="https://bugs.webkit.org/show_bug.cgi?id=264624">https://bugs.webkit.org/show_bug.cgi?id=264624</a>

Reviewed by Chris Dumez.

Deploy smart pointers as warned by the clang static analyzer.

* Source/WebCore/dom/AbortSignal.cpp:
(WebCore::AbortSignal::throwIfAborted):
* Source/WebCore/dom/BroadcastChannel.cpp:
(WebCore::BroadcastChannel::MainThreadBridge::MainThreadBridge):
* Source/WebCore/dom/NodeRareData.h:
(WebCore::NodeListsNodeData::removeChildNodeList):
(WebCore::NodeListsNodeData::removeEmptyChildNodeList):
* Source/WebCore/dom/Range.cpp:
(WebCore::Range::processContents):
(WebCore::Range::surroundContents):
(WebCore::boundaryTextNodesSplit):

Canonical link: <a href="https://commits.webkit.org/270578@main">https://commits.webkit.org/270578@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/2ce7b108834bb997aedf2387b95ce78a8e8f5abb

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/25798 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/4405 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/27080 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/27896 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/23619 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/26083 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/6167 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/1838 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/23726 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/26048 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/3310 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/22229 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/28476 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/2930 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/23184 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/29252 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/23544 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/23566 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/27122 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/2948 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/1176 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/4337 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/3403 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/3310 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/3265 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/WebCore/dom/NodeRareData.h:
```diff
@@ -68,7 +68,8 @@ class NodeListsNodeData {
         ASSERT(m_childNodeList == list);
-        if (deleteThisAndUpdateNodeRareDataIfAboutToRemoveLastList(list->ownerNode()))
+        Ref ownerNode = list->ownerNode();
+        if (deleteThisAndUpdateNodeRareDataIfAboutToRemoveLastList(ownerNode))
```
Why not use `list->protectedOwnerNode()` here?

rniwa replied:
It doesn't exist and we can't define it in ChildNodeList.h because that would require the declaration of Node object, which is only forward declared there.

cdumez replied:
What you did is fine. We could also have:
1. added protectedOwnerNode() and define it out of line if we're not particularly concerned about perf
2. added protectedOwnerNode() and declared it inlined in a new ChildNodeListInlines.h if we're concerned about perf.

rniwa replied:
Yeah, I think we probably want to do (2) since these code paths are relatively hot but I don't think it's worth the cost of adding a new header file for now.

cdumez commented on Source/WebCore/dom/NodeRareData.h:
```diff
@@ -86,7 +87,8 @@ class NodeListsNodeData {
         ASSERT(m_emptyChildNodeList == list);
-        if (deleteThisAndUpdateNodeRareDataIfAboutToRemoveLastList(list->ownerNode()))
+        Ref ownerNode = list->ownerNode();
+        if (deleteThisAndUpdateNodeRareDataIfAboutToRemoveLastList(ownerNode))
```
ditto.

rniwa replied:
Ditto.

rniwa committed:
Deploy more smart pointers in AbortSignal.cpp, BroadcastChannel.cpp, NodeRareData.h, and Range.cpp
https://bugs.webkit.org/show_bug.cgi?id=264624

Reviewed by Chris Dumez.

Deploy smart pointers as warned by the clang static analyzer.

* Source/WebCore/dom/AbortSignal.cpp:
(WebCore::AbortSignal::throwIfAborted):
* Source/WebCore/dom/BroadcastChannel.cpp:
(WebCore::BroadcastChannel::MainThreadBridge::MainThreadBridge):
* Source/WebCore/dom/NodeRareData.h:
(WebCore::NodeListsNodeData::removeChildNodeList):
(WebCore::NodeListsNodeData::removeEmptyChildNodeList):
* Source/WebCore/dom/Range.cpp:
(WebCore::Range::processContents):
(WebCore::Range::surroundContents):
(WebCore::boundaryTextNodesSplit):

Canonical link: https://commits.webkit.org/270578@main
